### ISSUE TITLE
Cannot manually start timer on componentDidMount

### ISSUE THREAD

**Issue by mavoi09** [SupremeTechnopriest/react-idle-timer#276]
How can I manually start the timer on componentDidMount?  Nothing happens when I call this.idleTime.start() in the componentDidMount, but if i call it on componentDidUpdate the timer starts. Anyone has a clue why this is not working?
 
     componentDidMount() {
        this.idleTimer.start()
    }
    
        render() {
          return <>
              <IdleTimer
                  name="UserIdleTimeout"
                  ref={this.setRef}
                  timeout={2000}
                  promptTimeout={5000}
                  onPrompt={this.onPrompt}
                  onIdle={this.onIdle}
                  onAction={this.onAction}
                  onActive={this.onActive}
                  startManually={true}
                  stopOnIdle={true}
              />
  
              <IdleTimeOutModal
                  showModal={false}
              />
          </>
    }

**Comment by SupremeTechnopriest** [comment]
Starting on mount is the default behavior.  Is there a specific reason you want to do it this way?  Are you using v4 still? V5 has built in prompt controls. You might want to consider updating. 

https://idletimer.dev/docs/api/with-idle-timer

I'm not sure what `setRef` is doing in your example above, but assuming it's storing the ref in `this.idleTimer`, this should work fine.  I am no longer supporting v4.  If you want to update to V5 and create a code sandbox that recreates your issue, I will be able to investigate.

**Comment by mavoi09** [comment]
> Starting on mount is the default behavior. Is there a specific reason you want to do it this way? Are you using v4 still? V5 has built in prompt controls. You might want to consider updating.
> 
> https://idletimer.dev/docs/api/with-idle-timer
> 
> I'm not sure what `setRef` is doing in your example above, but assuming it's storing the ref in `this.idleTimer`, this should work fine. I am no longer supporting v4. If you want to update to V5 and create a code sandbox that recreates your issue, I will be able to investigate.

I'm using version 5.4.1. setRef is a setter for the reference of the timer (this.idleTimer = ref). The idea is that  I want to start(or to stop it) the timer only if the user is logged in:
componentDidMount(){
    if(this.props.userIsLoggedIn){
        this.idletTImer.start();
    }
}

None of the methods, start(), stop(), activate() are working when they called on componendtDidMount. Guess I can do some workaround to render the IdleTimer only if the user is logged in and set startManually false.

**Comment by SupremeTechnopriest** [comment]
Are you using the higher order component to create your `IdleTimer` component?  V5 does not expose a component by default anymore.  Can you show me the code for your HOC?

**Comment by mavoi09** [comment]
> Are you using the higher order component to create your `IdleTimer` component? V5 does not expose a component by default anymore. Can you show me the code for your HOC?

import React from 'react';
import {withIdleTimer} from 'react-idle-timer';

const IdleTimer = withIdleTimer(() => {return <div></div>})


export class IdleTimerComponent extends React.Component {
	

    componentDidMount() {
		console.log(this.idleTimer)
        this.idleTimer.start()
    }
    
    setRef = (ref) =>{this.idleTimer = ref}
    
    onPrompt = () => {
        console.log('promp')
    } 
       
    onIdle = () => {
        console.log('idle')
    }   
        
    onAction = () => {
        console.log('action')
    }      
      
    onActive = () => {
        console.log('active')
    }
    
    render() {
        return <div>
            <IdleTimer
                name="UserIdleTimeout"
                ref={this.setRef}
                timeout={2000}
                promptTimeout={5000}
                onPrompt={this.onPrompt}
                onIdle={this.onIdle}
                onAction={this.onAction}
                onActive={this.onActive}
                startManually={true}
                stopOnIdle={true}
            />
        </div>
    }
}





import { render} from 'react-dom';
import React from "react";
import {IdleTimerComponent} from "./IdleTimer";

export default class App extends React.Component {
  constructor(props) {
    super(props);
    
  }
	
 componentDidMount(){

 }


  render() {
	  return <IdleTimerComponent/>;
  }
}

render(<App />, document.getElementById("root"));

**Comment by SupremeTechnopriest** [comment]
Can you try using a class component for the hoc:

```js
class IdleTimerComponent extends Component {
  render () {
    return this.props.children
  }
}

export const IdleTimer = withIdleTimer(IdleTimerComponent)```

**Comment by mavoi09** [comment]
I also tried with a class component, same result(also this.props.children is undefined)

**Comment by SupremeTechnopriest** [comment]
Okay seems to be a race condition going on.

https://codesandbox.io/s/tender-brook-p9oiyb?file=/src/App.js

If you wrap your start in a set timeout that will get you going for now, but I will look into the race condition.

```js
setTimeout(() => {
  this.idleTimer.start();
});
```

**Comment by SupremeTechnopriest** [comment]
@mavoi09 This is resolved in `5.4.2`. You will no longer need the `setTimeout`.
